sqlobject/src/main/java/org/jdbi/v3/sqlobject/TransactionDecorator.java

```diff
@@ -21,23 +21,25 @@
 import org.jdbi.v3.core.transaction.TransactionCallback;
 import org.jdbi.v3.core.transaction.TransactionIsolationLevel;
 
-class PassThroughTransactionHandler implements Handler
+class TransactionDecorator implements Handler
 {
+    private final Handler delegate;
     private final TransactionIsolationLevel isolation;
-    private final PassThroughHandler delegate;
 
-    PassThroughTransactionHandler(Transaction tx)
+    TransactionDecorator(Handler delegate, Transaction tx)
     {
+        this.delegate = delegate;
         this.isolation = tx.value();
-        this.delegate = new PassThroughHandler();
     }
 
     @Override
     public Object invoke(Supplier<Handle> handle, SqlObjectConfig config, Object target, Object[] args, Method method) throws Exception
     {
         Handle h = handle.get();
+
         if (h.isInTransaction()) {
-            throw new TransactionException("Nested @Transaction detected - this is currently not supported.");
+            // Already in transaction. The outermost @Transaction method determines the transaction isolation level.
```

Comment:
I believe the common 4 are in increasing order of strictness, but I am not sure this means that it is safe to increase mid transaction.  Do databases even support that?  Also, things run under `REPEATABLE READ` or `SERIALIZABLE` may spuriously abort and need to be retried, which code expecting to run at `READ COMMITTED` would not be able to handle.  I'm worried that being too lenient here just doesn't make sense, and lean towards forbidding it unless we can come up with a sensible use case other than trying to make sloppy / broken code work.